[bug] Max TF pre-commit job is failing on IRSA roles

## Description
When the pre-commit jobs are running for PRs we're seeing errors from some of blueprints that seem related to the IRSA roles. 
example https://github.com/awslabs/data-on-eks/actions/runs/16998090932/job/48193560716?pr=889

## Versions

- Module version [Required]: 1.2.0

- Terraform version:
Terraform v1.12.2
on darwin_arm64
+ provider registry.terraform.io/alekc/kubectl v2.1.3
+ provider registry.terraform.io/hashicorp/aws v5.100.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.3.7
+ provider registry.terraform.io/hashicorp/helm v2.17.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.38.0
+ provider registry.terraform.io/hashicorp/null v3.2.4
+ provider registry.terraform.io/hashicorp/random v3.7.2
+ provider registry.terraform.io/hashicorp/time v0.13.1
+ provider registry.terraform.io/hashicorp/tls v4.1.0

## Reproduction Code [Required]

Create a new PR that modifies the terraform files to kick off the terraform validation job in the pre-commit jobs

## Expected behavior

Ideally all jobs pass and all blueprints are valid.

## Actual behavior

Errors from the example PR task linked above:

```
 'terraform init' failed, '/usr/bin/terraform validate' skipped: schedulers/terraform/aws-batch-eks
[...]
╷
│ Error: Unreadable module subdirectory
│ 
│ The directory
│ .terraform/modules/cloudwatch_irsa_role/modules/iam-role-for-service-accounts-eks
│ does not exist. The target submodule
│ modules/iam-role-for-service-accounts-eks does not exist within the target
│ module.
╵
╷
│ Error: Unreadable module subdirectory
│ 
│ The directory
│ .terraform/modules/ebs_csi_irsa_role/modules/iam-role-for-service-accounts-eks
│ does not exist. The target submodule
│ modules/iam-role-for-service-accounts-eks does not exist within the target
│ module.
╵


'terraform init' failed, '/usr/bin/terraform validate' skipped: streaming/spark-streaming/terraform
[...]

│ Error: Unreadable module subdirectory
│ 
│ The directory
│ .terraform/modules/producer_iam_role/modules/iam-role-for-service-accounts-eks
│ does not exist. The target submodule
│ modules/iam-role-for-service-accounts-eks does not exist within the target
│ module.
╵

Error: Process completed with exit code 1.
```

--- a/schedulers/terraform/aws-batch-eks/README.md
+++ b/schedulers/terraform/aws-batch-eks/README.md
@@ -23,8 +23,8 @@ Checkout the [documentation website](https://awslabs.github.io/data-on-eks/docs/
 
 | Name | Source | Version |
 |------|--------|---------|
-| <a name="module_cloudwatch_irsa_role"></a> [cloudwatch\_irsa\_role](#module\_cloudwatch\_irsa\_role) | terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks | >= 5.44 |
-| <a name="module_ebs_csi_irsa_role"></a> [ebs\_csi\_irsa\_role](#module\_ebs\_csi\_irsa\_role) | terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks | >= 5.44 |
+| <a name="module_cloudwatch_irsa_role"></a> [cloudwatch\_irsa\_role](#module\_cloudwatch\_irsa\_role) | terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks | ~> 5.55 |
+| <a name="module_ebs_csi_irsa_role"></a> [ebs\_csi\_irsa\_role](#module\_ebs\_csi\_irsa\_role) | terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks | ~> 5.55 |
 | <a name="module_eks"></a> [eks](#module\_eks) | terraform-aws-modules/eks/aws | ~> 20.0 |
 | <a name="module_eks_auth"></a> [eks\_auth](#module\_eks\_auth) | terraform-aws-modules/eks/aws//modules/aws-auth | ~> 20.0 |
 | <a name="module_eks_blueprints_addons"></a> [eks\_blueprints\_addons](#module\_eks\_blueprints\_addons) | aws-ia/eks-blueprints-addons/aws | ~> 1.2 |

--- a/schedulers/terraform/aws-batch-eks/addons.tf
+++ b/schedulers/terraform/aws-batch-eks/addons.tf
@@ -3,7 +3,7 @@
 #---------------------------------------------------------------
 module "ebs_csi_irsa_role" {
   source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
-  version = ">= 5.44"
+  version = "~> 5.55"
 
   role_name = join("_", [var.eks_cluster_name, "ebs-csi"])
 
@@ -23,7 +23,7 @@ module "ebs_csi_irsa_role" {
 #---------------------------------------------------------------
 module "cloudwatch_irsa_role" {
   source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
-  version = ">= 5.44"
+  version = "~> 5.55"
 
   role_name = join("_", [var.eks_cluster_name, "cloudwatch"])
 

--- a/streaming/spark-streaming/terraform/README.md
+++ b/streaming/spark-streaming/terraform/README.md
@@ -31,7 +31,7 @@ Checkout the [documentation website](https://awslabs.github.io/data-on-eks/docs/
 | <a name="module_eks"></a> [eks](#module\_eks) | terraform-aws-modules/eks/aws | ~> 19.15 |
 | <a name="module_eks_blueprints_addons"></a> [eks\_blueprints\_addons](#module\_eks\_blueprints\_addons) | aws-ia/eks-blueprints-addons/aws | ~> 1.2 |
 | <a name="module_eks_data_addons"></a> [eks\_data\_addons](#module\_eks\_data\_addons) | aws-ia/eks-data-addons/aws | 1.33.0 |
-| <a name="module_producer_iam_role"></a> [producer\_iam\_role](#module\_producer\_iam\_role) | terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks | n/a |
+| <a name="module_producer_iam_role"></a> [producer\_iam\_role](#module\_producer\_iam\_role) | terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks | ~> 5.55 |
 | <a name="module_s3_bucket"></a> [s3\_bucket](#module\_s3\_bucket) | terraform-aws-modules/s3-bucket/aws | ~> 3.0 |
 | <a name="module_spark_team_a_irsa"></a> [spark\_team\_a\_irsa](#module\_spark\_team\_a\_irsa) | aws-ia/eks-blueprints-addon/aws | ~> 1.0 |
 | <a name="module_vpc"></a> [vpc](#module\_vpc) | terraform-aws-modules/vpc/aws | ~> 5.0 |

--- a/streaming/spark-streaming/terraform/apps.tf
+++ b/streaming/spark-streaming/terraform/apps.tf
@@ -57,7 +57,8 @@ resource "aws_iam_policy" "consumer_s3_kafka" {
 }
 
 module "producer_iam_role" {
-  source = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 5.55"
 
   role_name = "producer-irsa"
 

